---
title: "Astrocyte markers - Initial results"
author: "Dante von Zuben e Jo√£o Cavalcante"
date: "`r Sys.setlocale('LC_TIME', 'C'); format(Sys.time(), '%d %B, %Y')`"
knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, encoding = encoding, output_dir = "reports/") })
output:
  html_document:
    toc: true
    toc_float: true
    toc_collapsed: false
    theme:
      bslib: true
      bootswatch: minty
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,  message = FALSE)

library(here)
library(DT)

make_datatable <- function(data, colnames) {
  data %>%
    datatable(
      .,
      extensions = "Buttons",
      rownames = FALSE,
      escape = FALSE,
      colnames = colnames,
      filter = 'top',
      options = list(
        pageLength = 5,
        scrollX = TRUE,
        dom = '<"d-flex justify-content-between"lBf>rtip',
        buttons = list(
          'copy',
          list(
            extend = 'collection',
            buttons = c('csv', 'excel'),
            text = 'Download'
          )
        )
      )
    )
}
```

## GenePlast results

COG identifiers for each gene were acquired from the STRINGdb v11.5 COG mappings file.
The rooting analysis was then performed for these COG IDs.

:::{.column-body-outset}
```{r geneplast-table}
geneplast_roots <-
  vroom::vroom(here("results/enraizamento_genes_astrocitos.csv"))

make_datatable(geneplast_roots,
               colnames = c("COG", "Root", "Clade", "Symbol", "ENSP"))
```
:::

## Orthologs

1-to-1 orthologs gathered through the Ensembl database. 
When an ortholog is not present, it couldn't be found.

:::{.column-body-outset}
```{r ortho-table}
orthologs <-
  vroom::vroom(here("data/orthologs.csv"))

make_datatable(orthologs,
               colnames = c("H. sapiens", "M. musculus", "P. vitticeps", "L. striata domestica"))
```
:::

## Future goals

### Selected datasets

- [PRJNA408230](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA408230) (DropSeq) - 2020
  - Pallium expression in Pogona vitticeps (lizard species)
  - 35 samples
  
- [GSE150486](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE150486) (10x) - 2020
  - Gene expression in two song-production related regions in Lonchura striata domestica (Bengali Finch)
  - 4 samples

- [GSE230116](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE230116) (10x) - 2023 
  - Astrocyte expression in mice (only WT samples will be used)
  - 4 samples

### Methodology

- Data will be acquired from [SRA](https://www.ncbi.nlm.nih.gov/sra/) through the [nf-core/fetchngs](https://github.com/nf-core/fetchngs) pipeline.

- Count matrices will be assembled through the [nf-core/scrnaseq](https://github.com/nf-core/scrnaseq) pipeline.

- Matrices will be then processed in the R programming language.

- Given download and processing time for all datasets, an expected timeline for preliminary results is the first week of November.

- All computational infrastructure will be provided by [NPAD/UFRN](http://npad.ufrn.br/).



